### DISCUSSION TITLE
tesseract brew version 4.1.x has been overwritten

### DISCUSSION THREAD

**Discussion by JohnDuncanScott** [Homebrew/discussions#2735] [Tap maintenance and Homebrew development] [answered]
**Current Behavior:**
I'm not sure where to log this issue, hence this post. Apologies if this is the wrong place or if I've misunderstood, I haven't been using brew for that long.

* Multiple libraries depend on tesseract, SikuliX in this case
* 5.0.0 has breaking changes meaning a user cannot simply just use the latest version
* For Mac specifically, you **must** use a package manager to install tesseract (either MacPort or Brew)
* Having 2 package managers installed on the same Mac is not recommended at all and error prone
* Brew, from various discussions on StackOverflow, does not really support using specific commits when installing packages. It might work, but is not recommended and is not easy to do from what I understand
* Brew does support having fixed versions, e.g. postgresql@11 (found via brew search postgresql)
* The version in brew has just been overwritten rather than using the brew versioning. This means the latest version in brew is 5.0.0, getting 4.1.x back requires going to a specific commit, which as stated above may cause issues

**Expected Behavior:**
* Run brew install tesseract@4.1.3

The problem is I have no idea who actually uploads these things to brew (I think it's a bot in this case, not sure who actually owns it), why some packages use proper versions whereas others just seem to ignore that and overwrite with latest, or how to really report these issues. I tried the tesseract github itself, but they said it's not their problem. I could report this to the bot account, but not sure if anyone would pay attention. What's the usual process for this? Why is versioning so inconsistent in brew? Is there some command I'm missing that would make it trivial? I'm just not sure what the best practice is in this scenario.

Any help would be appreciated :)

**Comment by SMillerDev** [comment, marked as answer]
You can use `brew extract` to maintain your own version of the tesseract formula.

**Reply by carlocab** [reply]
To elaborate a bit:

You can use `brew tap-new` to create a tap to extract your own version of `tesseract` to:
```
brew tap-new JohnDuncanScott/your-tap-name-here
```
Then, extract the version of `tesseract` that you want to your tap:
```
brew extract --version=4.1.3 tesseract JohnDuncanScott/your-tap-name-here
```

To avoid `brew link` errors, you may wish to make this formula keg-only by adding `keg_only :versioned_formula` to the extracted formula (try `brew edit JohnDuncanScott/your-tap-name/tesseract@4.1.3`.

This will allow you to install the version that you want with
```
brew install JohnDuncanScott/your-tap-name-here/tesseract@4.1.3
```
 
Here are some additional docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

**Reply by JohnDuncanScott** [reply]
Thanks very much, appreciate the help :)

**Reply by sensimevanidus** [reply]
Thank you @carlocab
